pkg/controller/resource_sync_task.go

```diff
@@ -504,7 +512,7 @@ func (st *resourceSyncTask) prepareDependencies(dependsOn []smith_v1.ResourceNam
 				return nil, errors.Wrapf(err, "error processing ServiceBinding %q", obj.Name)
 			}
 		}
-		dependencies[name] = dependency
+		dependencies[reference.Resource] = dependency
```

Comment:
As I noted above, we have `name` and `resource`. If I understood correctly, here we should use `name` if it was set, and fallback to `resource` otherwise.